server/src/main/java/org/opensearch/cluster/service/ClusterManagerThrottlingDefaults.java

```
@@ -0,0 +1,64 @@
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.REMOVE_INDEX_TEMPLATE_V2_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.PUT_PIPELINE_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.DELETE_PIPELINE_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.PUT_SEARCH_PIPELINE_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.DELETE_SEARCH_PIPELINE_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.CREATE_PERSISTENT_TASK_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.FINISH_PERSISTENT_TASK_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.REMOVE_PERSISTENT_TASK_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.UPDATE_TASK_STATE_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.CREATE_QUERY_GROUP_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.DELETE_QUERY_GROUP_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.UPDATE_QUERY_GROUP_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.PUT_SCRIPT_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.DELETE_SCRIPT_KEY, 200);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.PUT_REPOSITORY_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.DELETE_REPOSITORY_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.CREATE_SNAPSHOT_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.DELETE_SNAPSHOT_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.UPDATE_SNAPSHOT_STATE_KEY, 5000);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.RESTORE_SNAPSHOT_KEY, 50);
+        DEFAULT_THRESHOLDS.put(ClusterManagerTaskKeys.CLUSTER_REROUTE_API_KEY, 50);
+    }
+
+    public static int getDefaultThreshold(String taskKey) {
+        return DEFAULT_THRESHOLDS.getOrDefault(taskKey, DEFAULT_THRESHOLD_VALUE);
```

Do we want to keep default value of getOrDefault to be `MIN_THRESHOLD_VALUE` (-1) or `DEFAULT_THRESHOLD_VALUE`? This will determine the behavior of the throttling for new tasks on boarded to framework. 

We can add validation for that that whatever new tasks come, that needs to have some value in this map. For that maybe we can remove `getOrDefault` and throw the exception if that key is not present in Map.

---

Sure, will add changes to throw exception if threshold value doesn't exist in this map for a task.

---

Sure, will add changes to throw an exception. So, that someone registering a new tasks is aware of default throttling threshold and can configure the threshold accordingly if needed.